src/app/frontend/deploy/deployfromfile_controller.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Controller for the deploy from file directive.
+ *
+ * @final
+ */
+export default class DeployFromFileController {
+  /**
+   * @param {!angular.$q} $q
+   * @ngInject
+   */
+  constructor($q) {
+    /**
+     * It initializes the scope output parameter
+     *
+     * @export {!DeployFromFileController}
+     */
+    this.detail = this;
+
+    /** @export {string} */
+    this.fileName = '';
+
+    /** @private {string} */
+    this.file_ = '';
+
+    /** @private {!angular.$q} */
+    this.q_ = $q;
+  }
+
+  /**
+   * It updates the model for the selected file
+   * @param {HTMLInputElement} element
+   * @export
+   */
+  updateFile(element) {
+    let file = element.files[0];
+    this.fileName = file.name;
+    let reader = new FileReader();
+    reader.onloadend = (event) => { this.file_ = event.target.result; };
+    reader.readAsBinaryString(file);
+    let defer = this.q_.defer();
```

Comment:
Thank you. If you use separate directive with link function that feeds ngModelController and ng-model on the file input this would be very clean and "the way it should be done in Angular" :)